pkg/server/biz/scm/gitlab/gitlabv4.go

```diff
@@ -198,22 +211,31 @@ func (g *V4) CreateWebhook(repo string, webhook *scm.Webhook) error {
 	hook.URL = &webhook.URL
 
 	onwer, name := scm.ParseRepo(repo)
-	_, _, err := g.client.Projects.AddProjectHook(onwer+"/"+name, &hook)
-	return err
+	_, resp, err := g.client.Projects.AddProjectHook(onwer+"/"+name, &hook)
+	if err != nil {
+		if resp.StatusCode == 500 {
+			return cerr.ErrorSCMServerInternalError.Error(g.scmCfg.Server, err)
+		}
+		return err
+	}
+	return nil
 }
 
 // DeleteWebhook deletes webhook from specified repo.
 func (g *V4) DeleteWebhook(repo string, webhookURL string) error {
 	owner, name := scm.ParseRepo(repo)
-	hooks, _, err := g.client.Projects.ListProjectHooks(owner+"/"+name, nil)
+	hooks, resp, err := g.client.Projects.ListProjectHooks(owner+"/"+name, nil)
 	if err != nil {
+		if resp.StatusCode == 500 {
```

Comment:
Define a error convertor, like `ConvertK8sError`: https://github.com/caicloud/cyclone/pull/918/files#diff-5053a20c9441cba09f265d48948f0a6cL78

Reply:
I found Github has define error response, but gitlab does not have. We can first only do that for Github error.